### PR TITLE
TST: Update unreliable num precision component of test_binary_arith_ops (GH37328)

### LINKED ISSUE #37328: TST: eval test unreliable
Ran into an unrelated failure I haven't seen before in #37227. Occurs in pandas/tests/computation/test_eval.py:211

https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=45754&view=logs&j=89027c74-d1cd-5e5d-942f-e6a83ebe07fe&t=768a7acf-33bc-5b3d-16e1-da874611990b&l=92

```

E           numpy array values are different (2.0 %)
E           [left]:  [[10.885282508088554, 24.01653292315993, 2.5334764666276754, -26.20850153209755, -9.292195139788854], [4.631505078070531, 0.43305142455418066, 9.796254287558813, -0.817344630073512, -0.22979400652647], [16.964677878827604, 3.69159983714683, 0.16736723603895715, -13.866779975183137, -0.6547978574323233], [0.9332007475405953, 7.326122317479498, 2.2912333245159338, -0.9957313583066364, -0.21968895568980523], [1.2901449066742292, 1.5178122458579082, 0.3101899430985658, -0.3600640963287845, -3.8622009358387115], [1.7324531378555383, 1.6717678316305915, 0.39248288379342006, -2.259323953594114, -13.215192982180358], [60.71781357885021, 13.08256705151808, 3.306197592158367, -2.603163236030203, -0.5783506355776388], [47.71284995565457, 39.508260846448245, 0.16695497225705538, -7.39400467084908, -0.9437156402277913], [0.40861584564047, 611.3601636482858, 0.8350779591410551, -3.034557944041943, -0.33953697203938915], [46.62807235120426, 6.308023182021756, 5.01671850424403, -0.23877079736974396, -0.5460348132251863]]
E           [right]: [[10.885282508088554, 24.01653292315993, 2.5334764666276754, -26.20850153209755, -9.292195139788854], [4.631505078070531, 0.43305142455418066, 9.796254287558813, -0.817344630073512, -0.22979400652647], [16.964677878827608, 3.69159983714683, 0.16736723603895715, -13.866779975183137, -0.6547978574323233], [0.9332007475405953, 7.326122317479498, 2.2912333245159338, -0.9957313583066364, -0.21968895568980523], [1.2901449066742292, 1.5178122458579082, 0.3101899430985658, -0.3600640963287845, -3.8622009358387115], [1.7324531378555383, 1.6717678316305915, 0.39248288379342006, -2.259323953594114, -13.215192982180358], [60.71781357885021, 13.08256705151808, 3.306197592158367, -2.603163236030203, -0.5783506355776388], [47.71284995565457, 39.508260846448245, 0.16695497225705538, -7.39400467084908, -0.9437156402277913], [0.40861584564047, 611.3601636482858, 0.8350779591410551, -3.034557944041943, -0.33953697203938915], [46.62807235120426, 6.308023182021756, 5.01671850424403, -0.23877079736974396, -0.5460348132251863]]
```

where one entry is `16.964677878827604` and the other is `16.964677878827608`. Haven't looked into it, but guessing this is a numerical precision issue.

### PR DESCRIPTION
**Pull request by avinashpancham**
- [x] closes #37328 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by avinashpancham** [comment]
Seems like a numerical precision issue indeed.

Comparing the values: 16.964677878827604 and 16.964677878827608 I get the following tolerance levels:
Max absolute difference: 3.55271368e-15
Max relative difference: 2.09418281e-16

Therefore I changed: `tm.assert_numpy_array_equal(result.values, expected)` -> `tm.assert_almost_equal(result.values, expected, atol=1e-15, rtol=1e-16)`

**Comment by avinashpancham** [comment]
I see that this issue is not limited to those values, I also get this error on #36838, where there's a mismatch between 0.8129387113107932 and 0.8129387113107933

The proposed change would catch it, but maybe we will need to relax the tolerances more to make it generic.

**Comment by dsaxton** [comment]
@avinashpancham Merging master should fix the linting error

**Comment by rhshadrach** [comment]
xref #13339 which changed the first assert in `check_binary_arith_op` to `assert_almost_equal`. Make sense we would need to do the same here.

**Comment by avinashpancham** [comment]
@rhshadrach sorry but I dont understand what you mean. Could you maybe give some more explanation?

**Comment by avinashpancham** [comment]
@dsaxton thanks that solved it indeed

**Comment by avinashpancham** [comment]
> @rhshadrach sorry but I dont understand what you mean. Could you maybe give some more explanation?

Now that I read it again it looks more like a general comment, instead of feedback. My bad.

**Comment by avinashpancham** [comment]
CI failed due to: `RuntimeError: can't start new thread`. Would it be possible for someone to restart the CI?

**Comment by simonjayhawkins** [comment]
> CI failed due to: `RuntimeError: can't start new thread`. Would it be possible for someone to restart the CI?

restarted.

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by jreback** [comment]
cc @jbrockmendel this seems reasonble to me.

File: pandas/tests/computation/test_eval.py [hunk 1]
```diff
@@ -298,10 +298,9 @@ def check_alignment(self, result, nlhs, ghs, op):
             # TypeError, AttributeError: series or frame with scalar align
             pass
         else:
-
             # direct numpy comparison
             expected = self.ne.evaluate(f"nlhs {op} ghs")
-            tm.assert_numpy_array_equal(result.values, expected)
+            tm.assert_almost_equal(result.values, expected)
```
Comment (jreback) [inline pandas/tests/computation/test_eval.py:303]: can you add a comment why we are doing this and reference to this issue

**Review by jreback — CHANGES_REQUESTED** [review]
pls merge master and ping on green.

File: pandas/tests/computation/test_eval.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/computation/test_eval.py:303]: yah i think this makes sense, but should have a TODO indicating that it should be equality and we had to punt

**Review by jreback — APPROVED** [review]

**Comment by avinashpancham** [comment]
@jreback merged master and CI is greenish. Errors are memory/OS errors unrelated to this change, do we want to run this again or is this fine?

**Comment by jreback** [comment]
thanks @avinashpancham

**Pull request merged**